### ISSUE TITLE
[Enhancement] Add Cache-Control to different part of the core of Invidious

### ISSUE THREAD

**Issue by unixfox** [iv-org/invidious#3046]
**Is your enhancement request related to a problem? Please describe.**
In order to reduce the load on Invidious, we should add come cache control so that a client doesn't request again to the server the information if it already has it in memory.

**Describe the solution you'd like**
On any endpoint we can, we should add a [Cache-Control](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control) header with the appropriate duration.

We can take a look at what duration each endpoint in Piped are cached: https://github.com/TeamPiped/Piped-Backend/blob/master/src/main/java/me/kavin/piped/ServerLauncher.java#L78

**Comment by SamantazFox** [comment]
Yup, definitely! I didn't realize until now that only our assets have cache control.
